Mathlib/Combinatorics/SimpleGraph/Basic.lean

```diff
@@ -1216,6 +1216,47 @@ theorem DeleteFar.mono (h : G.DeleteFar p r₂) (hr : r₁ ≤ r₂) : G.DeleteF
 
 end DeleteFar
 
+/-! ## Vertex replacement -/
+
+
+/-- The graph formed by replacing `t` with a copy of `s`, by changing `t`'s neighbours to match.
+The `s-t` edge is removed if present. -/
+def replaceVertex (s t : V) : SimpleGraph V where
+  Adj := G.Adj \ Sym2.ToRel (G.incidenceSet t) ⊔ Sym2.ToRel {⟦(t, v)⟧ | v ∈ G.neighborSet s \ {t}}
```

Comment:
You could do
```lean
Adj u v :=
  if u = t then
    If v = t then
        False
    else
        G.Adj s v
  else
    If v = t then
        G.Adj u s
    else
        G.Adj u v
```
If you're lucky the proofs follow by case-bashing, but I have no idea if this is bettee than your approach.

Reply:
Ah, that's not quite right because I didn't handle the s-t edge

Reply:
Actually your if-else tree works once `[DecidableEq V]` is added – see latest commit – and all proofs shorten with this change. But `cliqueFree_of_replaceVertex_cliqueFree` is still quite long.